tests/integrationv2/utils.py

```diff
@@ -14,7 +16,7 @@ def get_expected_s2n_version(protocol, provider):
     tls12 is always chosen. This is true even when the requested
     protocol is less than tls12.
     """
-    if provider == S2N and protocol != Protocols.TLS13:
+    if (provider == S2N or provider == CriterionS2N) and protocol != Protocols.TLS13:
```

Comment:
You can decide if this is nicer or not, but since you made CriterionS2N a subclass of S2N, a check of `issubclass(provider, S2N)` would work here I think!

```
>>> class A:
...     pass
... 
>>> class B(A):
...     pass
... 
>>> class C:
...     pass
... 
>>> issubclass(B, A)
True
>>> issubclass(A, A)
True
>>> issubclass(C, A)
False
```